# Super Store Sales Analysis
## Analyse sales data of Super Store mart and identify opportunities to boost business growth.
Super Store is a small retail business located in the United States. They sell Furniture, Office Supplies and Technology products, and their customers are the mass Consumer, Corporate and Home Offices.
The [data set(in CSV)](https://www.kaggle.com/datasets/vivek468/superstore-dataset-final) contains sales, profit and geographical information of Super Store. [Dataset in Excel](https://community.tableau.com/s/question/0D54T00000CWeX8SAL/sample-superstore-sales-excelxls).
Our task is to analyse the sales data and identify weak areas and opportunities for "Super-Store" to boost business growth.

### Business Questions
- Which Category is Best Selling and Most Profitable?
- What are the Best Selling and Most Profitable Sub-Category?
- Which is the Top Selling Sub-Category?
- Which Customer Segment is Most Profitable?
- Which is the Preferred Ship Mode?
- Which Region is the Most Profitable?
- Which City has the Highest Number of Sales?

### Preparing the Environment
We will import the required libraries and read in the data set.
- Pandas — Data manipulation
- Matplotlib and Seaborn — Data visualisation

In [1]:
# Import libraries and alias for easy reading
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## About Dataset
### Context
With growing demands and cut-throat competitions in the market, a Superstore Giant is seeking your knowledge in understanding what works best for them. They would like to understand which products, regions, categories and customer segments they should target or avoid. 

// TODO
You can even take this a step further and try and build a Regression model to predict Sales or Profit.

### Metadata
- Row ID => Unique ID for each row.
- Order ID => Unique Order ID for each Customer.
- Order Date => Order Date of the product.
- Ship Date => Shipping Date of the Product.
- Ship Mode=> Shipping Mode specified by the Customer.
- Customer ID => Unique ID to identify each Customer.
- Customer Name => Name of the Customer.
- Segment => The segment where the Customer belongs.
- Country => Customer's country of residence.
- City =>  Customer's City of residence.
- State =>  Customer's state of residence.
- Postal Code => Postal Code of every Customer.
- Region => Region where the Customer belongs.
- Product ID => Unique ID of the Product.
- Category => Category of the product ordered.
- Sub-Category => Sub-Category of the product ordered.
- Product Name => Name of the Product.
- Sales => Sales of the Product.
- Quantity => Quantity of the Product.
- Discount => Discount provided.
- Profit => Profit/Loss incurred.

In [2]:
# Read data and set to a dataframe 
df = pd.read_excel('./Dataset/Superstore.xls')
df

Row ID        Order ID Order Date  Ship Date       Ship Mode  \
0          1  CA-2016-152156 2016-11-08 2016-11-11    Second Class   
1          2  CA-2016-152156 2016-11-08 2016-11-11    Second Class   
2          3  CA-2016-138688 2016-06-12 2016-06-16    Second Class   
3          4  US-2015-108966 2015-10-11 2015-10-18  Standard Class   
4          5  US-2015-108966 2015-10-11 2015-10-18  Standard Class   
...      ...             ...        ...        ...             ...   
9989    9990  CA-2014-110422 2014-01-21 2014-01-23    Second Class   
9990    9991  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9991    9992  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9992    9993  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9993    9994  CA-2017-119914 2017-05-04 2017-05-09    Second Class   

     Customer ID     Customer Name    Segment        Country             City  \
0       CG-12520       Claire Gute   Consumer  United States        Henderson   
1       CG-12520       Claire Gute   Consumer  United States        Henderson   
2       DV-13045   Darrin Van Huff  Corporate  United States      Los Angeles   
3       SO-20335    Sean O'Donnell   Consumer  United States  Fort Lauderdale   
4       SO-20335    Sean O'Donnell   Consumer  United States  Fort Lauderdale   
...          ...               ...        ...            ...              ...   
9989    TB-21400  Tom Boeckenhauer   Consumer  United States            Miami   
9990    DB-13060       Dave Brooks   Consumer  United States       Costa Mesa   
9991    DB-13060       Dave Brooks   Consumer  United States       Costa Mesa   
9992    DB-13060       Dave Brooks   Consumer  United States       Costa Mesa   
9993    CC-12220      Chris Cortes   Consumer  United States      Westminster   

      ... Postal Code  Region       Product ID         Category Sub-Category  \
0     ...       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1     ...       42420   South  FUR-CH-10000454        Furniture       Chairs   
2     ...       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3     ...       33311   South  FUR-TA-10000577        Furniture       Tables   
4     ...       33311   South  OFF-ST-10000760  Office Supplies      Storage   
...   ...         ...     ...              ...              ...          ...   
9989  ...       33180   South  FUR-FU-10001889        Furniture  Furnishings   
9990  ...       92627    West  FUR-FU-10000747        Furniture  Furnishings   
9991  ...       92627    West  TEC-PH-10003645       Technology       Phones   
9992  ...       92627    West  OFF-PA-10004041  Office Supplies        Paper   
9993  ...       92683    West  OFF-AP-10002684  Office Supplies   Appliances   

                                           Product Name     Sales  Quantity  \
0                     Bush Somerset Collection Bookcase  261.9600         2   
1     Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2     Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3         Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                        Eldon Fold 'N Roll Cart System   22.3680         2   
...                                                 ...       ...       ...   
9989                             Ultra Door Pull Handle   25.2480         3   
9990  Tenex B1-RE Series Chair Mats for Low Pile Car...   91.9600         2   
9991                              Aastra 57i VoIP phone  258.5760         2   
9992  It's Hot Message Books with Stickers, 2 3/4" x 5"   29.6000         4   
9993  Acco 7-Outlet Masterpiece Power Center, Wihtou...  243.1600         2   

      Discount    Profit  
0         0.00   41.9136  
1         0.00  219.5820  
2         0.00    6.8714  
3         0.45 -383.0310  
4         0.20    2.5164  
...        ...       ...  
9989      0.20    4.1028  
9990      0.00   15.6332  
9991      0.20   19.3932  
9992      0.

### Data Exploration
Let’s have a look at the data using df.head() and df.tail() function.

In [5]:
# Preview first 5 rows of data set
df.head() # defaults to 5

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  

[5 rows x 21 columns]

In [4]:
# Preview last 5 rows of data set
df.tail() # defaults to 5

Row ID        Order ID Order Date  Ship Date       Ship Mode  \
9989    9990  CA-2014-110422 2014-01-21 2014-01-23    Second Class   
9990    9991  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9991    9992  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9992    9993  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9993    9994  CA-2017-119914 2017-05-04 2017-05-09    Second Class   

     Customer ID     Customer Name   Segment        Country         City  ...  \
9989    TB-21400  Tom Boeckenhauer  Consumer  United States        Miami  ...   
9990    DB-13060       Dave Brooks  Consumer  United States   Costa Mesa  ...   
9991    DB-13060       Dave Brooks  Consumer  United States   Costa Mesa  ...   
9992    DB-13060       Dave Brooks  Consumer  United States   Costa Mesa  ...   
9993    CC-12220      Chris Cortes  Consumer  United States  Westminster  ...   

     Postal Code  Region       Product ID         Category Sub-Category  \
9989       33180   South  FUR-FU-10001889        Furniture  Furnishings   
9990       92627    West  FUR-FU-10000747        Furniture  Furnishings   
9991       92627    West  TEC-PH-10003645       Technology       Phones   
9992       92627    West  OFF-PA-10004041  Office Supplies        Paper   
9993       92683    West  OFF-AP-10002684  Office Supplies   Appliances   

                                           Product Name    Sales  Quantity  \
9989                             Ultra Door Pull Handle   25.248         3   
9990  Tenex B1-RE Series Chair Mats for Low Pile Car...   91.960         2   
9991                              Aastra 57i VoIP phone  258.576         2   
9992  It's Hot Message Books with Stickers, 2 3/4" x 5"   29.600         4   
9993  Acco 7-Outlet Masterpiece Power Center, Wihtou...  243.160         2   

      Discount   Profit  
9989       0.2   4.1028  
9990       0.0  15.6332  
9991       0.2  19.3932  
9992       0.0  13.3200  
9993       0.0  72.9480  

[5 rows x 21 columns]

In [6]:
# Show more rows in head() or tail()
df.head(8) # parameter to show n rows

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
5       6  CA-2014-115812 2014-06-09 2014-06-14  Standard Class    BH-11710   
6       7  CA-2014-115812 2014-06-09 2014-06-14  Standard Class    BH-11710   
7       8  CA-2014-115812 2014-06-09 2014-06-14  Standard Class    BH-11710   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
5  Brosina Hoffman   Consumer  United States      Los Angeles  ...   
6  Brosina Hoffman   Consumer  United States      Los Angeles  ...   
7  Brosina Hoffman   Consumer  United States      Los Angeles  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   
5       90032    West  FUR-FU-10001487        Furniture  Furnishings   
6       90032    West  OFF-AR-10002833  Office Supplies          Art   
7       90032    West  TEC-PH-10002275       Technology       Phones   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   
5  Eldon Expressions Wood and Plastic Desk Access...   48.8600         7   
6                                         Newell 322    7.2800         4   
7                     Mitel 5320 IP Phone VoIP phone  907.1520         6   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  
5      0.00   14.1694  
6      0.00    1.9656  
7      0.20   90.7152  

[8 rows x 21 columns]

You can easily tell that there is a mix of categorical, geographical and numerical variables.
Each row represents an order of an item and the corresponding quantity, sales, discount and profit. There is also the mode of shipment, customer segment and geographical aspects.
Next, we will use the **df.shape, df.info() and df.describe()** to get more information.

In [7]:
# Shape of data set
df.shape # This is an attribute not a method

(9994, 21)

In [8]:
df.shape() # This is an attribute not a method, hence error here

TypeError: 'tuple' object is not callable

In [9]:
# Summarised information of data set
df.info() # this is kind of a meta function

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Row ID         9994 non-null   int64         
 1   Order ID       9994 non-null   object        
 2   Order Date     9994 non-null   datetime64[ns]
 3   Ship Date      9994 non-null   datetime64[ns]
 4   Ship Mode      9994 non-null   object        
 5   Customer ID    9994 non-null   object        
 6   Customer Name  9994 non-null   object        
 7   Segment        9994 non-null   object        
 8   Country        9994 non-null   object        
 9   City           9994 non-null   object        
 10  State          9994 non-null   object        
 11  Postal Code    9994 non-null   int64         
 12  Region         9994 non-null   object        
 13  Product ID     9994 non-null   object        
 14  Category       9994 non-null   object        
 15  Sub-Category   9994 n

There are 9,994 rows (***excluding*** *header*) and 13 columns. Data type is correct and matches the corresponding values.

**df.describe()** generates descriptive statistics. Descriptive statistics include those that summarize the central tendency, dispersion and shape of a dataset’s distribution, excluding NaN values. Analyzes both numeric and object series, as well as DataFrame column sets of mixed data types. The output will vary depending on what is provided.

We will talk a bit more about this later.

In [11]:
# Calculates some statistical data like percentile, mean and std of the numerical values 
df.describe()

Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000

### Data Cleaning
Before we start with the analysis, we must first clean the data or “scrub the dirt.”
For this analysis, we will look at the more common issues such as missing and duplicate data.
<br/>
#### Handling Missing Values
Here, we will find out whether there is Null value.

In [13]:
# Find the number of null values for all columns
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

The result shows that the data set does not have any null value.
<br/>
### Duplicate Data
Then, we will find out whether there is duplicate data.

In [15]:
# Find the number of duplicate data
df.duplicated().sum()

0

The Result shows that there are **0** duplicated rows. If it had any duplicates, we could have used another function to view the duplicated data.

In [16]:
# Show the duplicated rows
df[df.duplicated(keep = 'last')]

Empty DataFrame
Columns: [Row ID, Order ID, Order Date, Ship Date, Ship Mode, Customer ID, Customer Name, Segment, Country, City, State, Postal Code, Region, Product ID, Category, Sub-Category, Product Name, Sales, Quantity, Discount, Profit]
Index: []

[0 rows x 21 columns]

We could then run the drop_duplicates() function to drop the duplicated rows.

In [17]:
# Drop the duplicated rows
df.drop_duplicates(inplace = True)

# Find the number of rows and columns
df.shape

(9994, 21)

Also, it is a good practice to name dataframes with some context and avoid using a generic name like df. It helps in working with multiple datasets. 

In [19]:
superstore = df

### Calculated Field
Now, we will create a calculated field for Profit Margin. The formula is (Profit / Sales) * 100.

In [20]:
superstore['Profit Margin %'] = (superstore.Profit / superstore.Sales) * 100
superstore

Row ID        Order ID Order Date  Ship Date       Ship Mode  \
0          1  CA-2016-152156 2016-11-08 2016-11-11    Second Class   
1          2  CA-2016-152156 2016-11-08 2016-11-11    Second Class   
2          3  CA-2016-138688 2016-06-12 2016-06-16    Second Class   
3          4  US-2015-108966 2015-10-11 2015-10-18  Standard Class   
4          5  US-2015-108966 2015-10-11 2015-10-18  Standard Class   
...      ...             ...        ...        ...             ...   
9989    9990  CA-2014-110422 2014-01-21 2014-01-23    Second Class   
9990    9991  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9991    9992  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9992    9993  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9993    9994  CA-2017-119914 2017-05-04 2017-05-09    Second Class   

     Customer ID     Customer Name    Segment        Country             City  \
0       CG-12520       Claire Gute   Consumer  United States        Henderson   
1       CG-12520       Claire Gute   Consumer  United States        Henderson   
2       DV-13045   Darrin Van Huff  Corporate  United States      Los Angeles   
3       SO-20335    Sean O'Donnell   Consumer  United States  Fort Lauderdale   
4       SO-20335    Sean O'Donnell   Consumer  United States  Fort Lauderdale   
...          ...               ...        ...            ...              ...   
9989    TB-21400  Tom Boeckenhauer   Consumer  United States            Miami   
9990    DB-13060       Dave Brooks   Consumer  United States       Costa Mesa   
9991    DB-13060       Dave Brooks   Consumer  United States       Costa Mesa   
9992    DB-13060       Dave Brooks   Consumer  United States       Costa Mesa   
9993    CC-12220      Chris Cortes   Consumer  United States      Westminster   

      ... Region       Product ID         Category Sub-Category  \
0     ...  South  FUR-BO-10001798        Furniture    Bookcases   
1     ...  South  FUR-CH-10000454        Furniture       Chairs   
2     ...   West  OFF-LA-10000240  Office Supplies       Labels   
3     ...  South  FUR-TA-10000577        Furniture       Tables   
4     ...  South  OFF-ST-10000760  Office Supplies      Storage   
...   ...    ...              ...              ...          ...   
9989  ...  South  FUR-FU-10001889        Furniture  Furnishings   
9990  ...   West  FUR-FU-10000747        Furniture  Furnishings   
9991  ...   West  TEC-PH-10003645       Technology       Phones   
9992  ...   West  OFF-PA-10004041  Office Supplies        Paper   
9993  ...   West  OFF-AP-10002684  Office Supplies   Appliances   

                                           Product Name     Sales Quantity  \
0                     Bush Somerset Collection Bookcase  261.9600        2   
1     Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400        3   
2     Self-Adhesive Address Labels for Typewriters b...   14.6200        2   
3         Bretford CR4500 Series Slim Rectangular Table  957.5775        5   
4                        Eldon Fold 'N Roll Cart System   22.3680        2   
...                                                 ...       ...      ...   
9989                             Ultra Door Pull Handle   25.2480        3   
9990  Tenex B1-RE Series Chair Mats for Low Pile Car...   91.9600        2   
9991                              Aastra 57i VoIP phone  258.5760        2   
9992  It's Hot Message Books with Stickers, 2 3/4" x 5"   29.6000        4   
9993  Acco 7-Outlet Masterpiece Power Center, Wihtou...  243.1600        2   

      Discount    Profit  Profit Margin %  
0         0.00   41.9136            16.00  
1         0.00  219.5820            30.00  
2         0.00    6.8714            47.00  
3         0.45 -383.0310           -40.00  
4         0.20    2.5164            11.25  
...        ...       ...              ...  
9989      0.20    4.1028            16.25  
9990      0.00   15.6332            17.00  
9991      0.20   19.3932             7.50  
9992      

Observe the last column above.
Now that the data set has been scrubbed, we can proceed with some statistics analysis!
<br/>
### Descriptive Statistics
Here, we will do a descriptive statistical analysis. We use df.describe() and assign ‘include = ‘all’ to ensure that categorical features are also included in the output.

In [24]:
# Get descriptive statistics summary
superstore.describe(include = "all")

C:\Users\manan\AppData\Local\Temp\ipykernel_10788\754186322.py:2: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  superstore.describe(include = "all")
C:\Users\manan\AppData\Local\Temp\ipykernel_10788\754186322.py:2: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  superstore.describe(include = "all")


Row ID        Order ID           Order Date            Ship Date  \
count   9994.000000            9994                 9994                 9994   
unique          NaN            5009                 1237                 1334   
top             NaN  CA-2017-100111  2016-09-05 00:00:00  2015-12-16 00:00:00   
freq            NaN              14                   38                   35   
first           NaN             NaN  2014-01-03 00:00:00  2014-01-07 00:00:00   
last            NaN             NaN  2017-12-30 00:00:00  2018-01-05 00:00:00   
mean    4997.500000             NaN                  NaN                  NaN   
std     2885.163629             NaN                  NaN                  NaN   
min        1.000000             NaN                  NaN                  NaN   
25%     2499.250000             NaN                  NaN                  NaN   
50%     4997.500000             NaN                  NaN                  NaN   
75%     7495.750000             NaN                  NaN                  NaN   
max     9994.000000             NaN                  NaN                  NaN   

             Ship Mode Customer ID  Customer Name   Segment        Country  \
count             9994        9994           9994      9994           9994   
unique               4         793            793         3              1   
top     Standard Class    WB-21850  William Brown  Consumer  United States   
freq              5968          37             37      5191           9994   
first              NaN         NaN            NaN       NaN            NaN   
last               NaN         NaN            NaN       NaN            NaN   
mean               NaN         NaN            NaN       NaN            NaN   
std                NaN         NaN            NaN       NaN            NaN   
min                NaN         NaN            NaN       NaN            NaN   
25%                NaN         NaN            NaN       NaN            NaN   
50%                NaN         NaN            NaN       NaN            NaN   
75%                NaN         NaN            NaN       NaN            NaN   
max                NaN         NaN            NaN       NaN            NaN   

                 City  ... Region       Product ID         Category  \
count            9994  ...   9994             9994             9994   
unique            531  ...      4             1862                3   
top     New York City  ...   West  OFF-PA-10001970  Office Supplies   
freq              915  ...   3203               19             6026   
first             NaN  ...    NaN              NaN              NaN   
last              NaN  ...    NaN              NaN              NaN   
mean              NaN  ...    NaN              NaN              NaN   
std               NaN  ...    NaN              NaN              NaN   
min               NaN  ...    NaN              NaN              NaN   
25%               NaN  ...    NaN              NaN              NaN   
50%               NaN  ...    NaN              NaN              NaN   
75%               NaN  ...    NaN              NaN              NaN   
max               NaN  ...    NaN              NaN              NaN   

       Sub-Category     Product Name         Sales     Quantity     Discount  \
count          9994             9994   9994.000000  9994.000000  9994.000000   
unique           17             1850           NaN          NaN          NaN   
top         Binders  Staple envelope           NaN          NaN          NaN   
freq           1523               48           NaN          NaN          NaN   
first           NaN              NaN           NaN          NaN          NaN   
last            NaN              NaN           NaN          NaN          NaN   
mean            NaN              NaN    229.858001     3.789574     0.156203   
std             NaN              NaN    623.245101     2.225110     0.206452   
min             NaN              NaN      0.444000     1.000000     0.

You will see ‘NaN’ in some of the categorical columns and that’s perfectly fine. Categorical values are not meant to have calculations performed on them, so we can ignore those. These are commonly Text fields, hence NaN: Not a Number. Observe the difference from the output earlier of the same method without include=="all" parameter.

What we want to focus on is the unique count and frequency of the categorical features such as
- There are 4 ship modes and 3 customer segments. Nearly half of the orders are from the Consumer segment using Standard Class shipment.
- The store carries 3 category of items with 17 "sub-category." 60% of orders are for Office Supplies, and at least 15 % of purchases are for Binders.

While for numerical parameters, it’s interesting to note that 
- 75% of orders makes at least 36% profit margin.
- Loss-making orders can go up to 275% losses. We must place emphasis on these loss-making sales to cut them off.

We will perform some analysis below to test these observations.

### Exploratory Data Analysis
1. Which Category is Best Selling and Most Profitable?
Best Selling means looking at the Sales and Most Profitable is referring to the Profit. We will group the Sales, Profit and also Quantity by Category and run the df.group().sum and pd.DataFrame() functions.